clients/ui/frontend/src/app/pages/modelCatalog/components/globalFilters/MaxLatencyFilter.tsx

```diff
@@ -0,0 +1,377 @@
+    },
+    [filterOptions],
+  );
+
+  // Local state for the filter configuration (persistent state)
+  const [appliedFilter, setAppliedFilter] = React.useState<LatencyFilterState>(() => {
+    // Initialize with first available options to ensure consistency
+    const firstAvailableMetric = availableMetrics.length > 0 ? availableMetrics[0].value : 'E2E';
+    const firstAvailablePercentile = getAvailablePercentiles(firstAvailableMetric);
+    const defaultPercentile =
+      firstAvailablePercentile.length > 0 ? firstAvailablePercentile[0].value : 'P90';
+
+    return {
+      metric: firstAvailableMetric,
+      percentile: defaultPercentile,
+      value: 30, // Reasonable default within typical TTFT range
+    };
+  });
+
+  // Working state while editing the filter
+  const [localFilter, setLocalFilter] = React.useState<LatencyFilterState>(appliedFilter);
+
+  const hasActiveFilter = savedFilterValue !== undefined;
+
+  const getDisplayText = (): string => {
+    if (hasActiveFilter) {
+      // When there's an active filter, show the full specification with actual selected values
+      return `Max latency: ${appliedFilter.metric} | ${appliedFilter.percentile} | ${savedFilterValue}ms`;
+    }
+    return 'Max latency';
+  };
+
+  const handleApplyFilter = () => {
+    // Store the current local filter values as the applied filter
+    setAppliedFilter(localFilter);
+    setSavedFilterValue(localFilter.value);
+    setIsOpen(false);
+  };
+
+  const handleReset = () => {
+    // Use first available options instead of hardcoded defaults
+    const firstAvailableMetric = availableMetrics.length > 0 ? availableMetrics[0].value : 'E2E';
+    const firstAvailablePercentile = getAvailablePercentiles(firstAvailableMetric);
+    const defaultPercentile =
+      firstAvailablePercentile.length > 0 ? firstAvailablePercentile[0].value : 'P90';
+
+    const defaultFilter: LatencyFilterState = {
+      metric: firstAvailableMetric,
+      percentile: defaultPercentile,
+      value: Math.min(maxValue, 100), // Use calculated maxValue but cap at reasonable level
+    };
+
+    setSavedFilterValue(undefined);
+    setAppliedFilter(defaultFilter);
+    setLocalFilter(defaultFilter);
+    setIsOpen(false);
+  };
+
+  // Calculate min/max latency values from performance artifacts
+  const { minValue, maxValue } = React.useMemo((): { minValue: number; maxValue: number } => {
+    if (performanceArtifacts.length === 0) {
+      return { minValue: 20, maxValue: 893 }; // Default values when no artifacts
+    }
+
+    // Get all latency values for the currently selected metric/percentile
+    const fieldName = getLatencyFieldName(appliedFilter.metric, appliedFilter.percentile);
+    const latencyValues = performanceArtifacts
+      .map((artifact) =>
+        getDoubleValue(
+          artifact.customProperties,
+          // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+          fieldName as keyof typeof artifact.customProperties,
```

Comment:
We could probably narrow the return type of getLatencyFieldName so that this assertion isn't necessary